evm-tests/jsontests/src/state.rs

```diff
@@ -47,76 +47,74 @@ pub struct StateTestsDump {
 	pub access_list: Vec<(H160, Vec<H256>)>,
 }
 
-#[allow(unused_variables)]
-impl StateTestsDump {
-	pub fn set_state(&mut self, state: &BTreeMap<H160, MemoryAccount>) {
-		#[cfg(feature = "dump-state")]
-		{
-			self.state = state.clone();
-		}
+trait StateTestsDumper {
+	fn set_state(&mut self, _state: &BTreeMap<H160, MemoryAccount>) {}
+	fn set_used_gas(&mut self, _used_gas: u64) {}
+	fn set_vicinity(&mut self, _vicinity: &MemoryVicinity) {}
+	fn set_tx_data(
+		&mut self,
+		_to: &H160,
+		_value: U256,
+		_data: Vec<u8>,
+		_gas_limit: u64,
+		_access_list: Vec<(H160, Vec<H256>)>,
+	) {
 	}
+	fn set_caller_secret_key(&mut self, _caller_secret_key: &H256) {}
+	fn set_state_hash(&mut self, _state_hash: &H256) {}
+	fn set_result_state(&mut self, _state: &BTreeMap<H160, MemoryAccount>) {}
+	#[allow(dead_code)]
```

Comment:
Is there a reason for that?

Reply:
The compiler noticed that it is not used anywhere. So, I added it to make the compiler happy.